[Bug]: hop-gui hangs and does not open pop-up windows

### Apache Hop version?

2.9

### Java version?

java 11.0.23 2024-04-16 LTS

### Operating system

Windows

### What happened?

The problem occurs every time I leave hop-gui unused for a while. If it remains open without use, then when I try to edit any action or create a new pipeline, nothing happens. The only solution is to restart hop-gui, and when I close it, it throws the following error in the cmd.

"Serious error detected in the Hop GUI: Failed to execute runnable (org.eclipse.swt.SWTException: Widget is disposed)
org.eclipse.swt.SWTException: Failed to execute runnable (org.eclipse.swt.SWTException: Widget is disposed)
        at org.eclipse.swt.SWT.error(SWT.java:4918)
        at org.eclipse.swt.SWT.error(SWT.java:4833)
        at org.eclipse.swt.widgets.Synchronizer.runAsyncMessages(Synchronizer.java:135)
        at org.eclipse.swt.widgets.Display.runAsyncMessages(Display.java:4029)
        at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3645)
        at org.eclipse.swt.widgets.Display.release(Display.java:3720)
        at org.eclipse.swt.graphics.Device.dispose(Device.java:324)
        at org.apache.hop.ui.hopgui.HopGui.open(HopGui.java:488)
        at org.apache.hop.ui.hopgui.HopGui.main(HopGui.java:352)
        Suppressed: org.eclipse.swt.SWTException: Failed to execute runnable (org.eclipse.swt.SWTException: Widget is disposed)
                ... 9 more
        Caused by: org.eclipse.swt.SWTException: Widget is disposed
                at org.eclipse.swt.SWT.error(SWT.java:4918)
                at org.eclipse.swt.SWT.error(SWT.java:4833)
                at org.eclipse.swt.SWT.error(SWT.java:4804)
                at org.eclipse.swt.widgets.Widget.error(Widget.java:450)
                at org.eclipse.swt.widgets.Widget.checkWidget(Widget.java:369)
                at org.eclipse.swt.widgets.Dialog.checkParent(Dialog.java:169)
                at org.eclipse.swt.widgets.Dialog.<init>(Dialog.java:131)
                at org.apache.hop.ui.core.dialog.ErrorDialog.<init>(ErrorDialog.java:78)
                at org.apache.hop.ui.hopgui.context.GuiContextUtil.lambda$handleActionSelection$0(GuiContextUtil.java:184)
                at org.eclipse.swt.widgets.RunnableLock.run(RunnableLock.java:40)
                at org.eclipse.swt.widgets.Synchronizer.runAsyncMessages(Synchronizer.java:132)
                ... 6 more
        Suppressed: org.eclipse.swt.SWTException: Failed to execute runnable (org.eclipse.swt.SWTException: Widget is disposed)
                ... 9 more
        Caused by: org.eclipse.swt.SWTException: Widget is disposed
                at org.eclipse.swt.SWT.error(SWT.java:4918)
                at org.eclipse.swt.SWT.error(SWT.java:4833)
                at org.eclipse.swt.SWT.error(SWT.java:4804)
                at org.eclipse.swt.widgets.Widget.error(Widget.java:450)
                at org.eclipse.swt.widgets.Widget.checkWidget(Widget.java:369)
                at org.eclipse.swt.widgets.Dialog.checkParent(Dialog.java:169)
                at org.eclipse.swt.widgets.Dialog.<init>(Dialog.java:131)
                at org.apache.hop.ui.core.dialog.ErrorDialog.<init>(ErrorDialog.java:78)
                at org.apache.hop.ui.hopgui.context.GuiContextUtil.lambda$handleActionSelection$0(GuiContextUtil.java:184)
                at org.eclipse.swt.widgets.RunnableLock.run(RunnableLock.java:40)
                at org.eclipse.swt.widgets.Synchronizer.runAsyncMessages(Synchronizer.java:132)
                ... 6 more
Caused by: org.eclipse.swt.SWTException: Widget is disposed
        at org.eclipse.swt.SWT.error(SWT.java:4918)
        at org.eclipse.swt.SWT.error(SWT.java:4833)
        at org.eclipse.swt.SWT.error(SWT.java:4804)
        at org.eclipse.swt.widgets.Widget.error(Widget.java:450)
        at org.eclipse.swt.widgets.Widget.checkWidget(Widget.java:369)
        at org.eclipse.swt.widgets.Dialog.checkParent(Dialog.java:169)
        at org.eclipse.swt.widgets.Dialog.<init>(Dialog.java:131)
        at org.apache.hop.ui.core.dialog.ErrorDialog.<init>(ErrorDialog.java:78)
        at org.apache.hop.ui.hopgui.context.GuiContextUtil.lambda$handleActionSelection$0(GuiContextUtil.java:184)
        at org.eclipse.swt.widgets.RunnableLock.run(RunnableLock.java:40)
        at org.eclipse.swt.widgets.Synchronizer.runAsyncMessages(Synchronizer.java:132)
        ... 6 more

Press any key to continue . . ."

### Issue Priority

Priority: 3

### Issue Component

Component: Hop Gui

--- a/ui/src/main/java/org/apache/hop/ui/hopgui/context/GuiContextUtil.java
+++ b/ui/src/main/java/org/apache/hop/ui/hopgui/context/GuiContextUtil.java
@@ -32,6 +32,7 @@ import org.apache.hop.ui.core.dialog.ErrorDialog;
 import org.apache.hop.ui.hopgui.HopGui;
 import org.apache.hop.ui.hopgui.ISingletonProvider;
 import org.apache.hop.ui.hopgui.ImplementationLoader;
+import org.apache.hop.ui.util.SwtErrorHandler;
 import org.eclipse.swt.widgets.Shell;
 
 public class GuiContextUtil {
@@ -181,7 +182,9 @@ public class GuiContextUtil {
                       IGuiActionLambda<?> actionLambda = selectedAction.getActionLambda();
                       actionLambda.executeAction(dialog.isShiftClicked(), dialog.isCtrlClicked());
                     } catch (Exception e) {
-                      new ErrorDialog(parent, "Error", "An error occurred executing action", e);
+                      if (!SwtErrorHandler.handleException(e)) {
+                        new ErrorDialog(parent, "Error", "An error occurred executing action", e);
+                      }
                     }
                   });
 

